lib/mshoplib/tests/MShop/Order/Manager/Base/StandardTest.php

```diff
@@ -708,9 +708,12 @@ public function testStoreExisting()
 			}
 		}
 
-		$newProducts = $newBasket->getProducts();
-		foreach( $basket->getProducts() as $key => $product ) {
-			$this->assertEquals( $product->getId(), $newProducts[$key]->getId() );
+		// streamline keys in fresh loaded order, see https://github.com/aimeos/aimeos-core/pull/280
+		$products = $basket->getProducts();
+		$productsKeys = $products->keys();
```

Comment:
A better way might be to use:
````
$basket->getProducts()->equals( $newbasket->getProducts() )
```
Can you check if that one-liner works? Same for addresses and services because they may also get reordered by plugins somewhere in the future.

Reply:
Woah, that would be some kind of magic. Sadly, `compare()` results in an error:

```
Array to string conversion

/var/customers/webs/belagode/test/aimeos-core/vendor/aimeos/map/src/Map.php:1107
```

Reply:
Unfortunately, `array_diff()` casts all elements to strings so it don't work but we can compare the IDs like before:
```
$basket->getProducts()->getId()->equals( $newbasket->getProducts()->getId() )
```

Reply:
Pretty neat, too. I guess I have to dig a bit deeper into map's functionality.